### DISCUSSION TITLE
How do I use fetch on a custom page to populate a relationship field?

### DISCUSSION THREAD

**Discussion by fronbow** [Laravel-Backpack/community-forum#1080] [Q&A (Help)]
Hey All

So, I've got a custom admin page that I want to be able to use a relationship field on:
![doorsheet-screen](https://github.com/Laravel-Backpack/community-forum/assets/1417531/72441e7e-78a5-4306-ba32-08aad87b3e40)

My idea is to use the form at the top of the page to populate the rows underneath but I can't seem to get the relationship field to populate?

(That current row is just a dummy, when in production it will be blank until clients are added using the form at the top of the page)

Given that it's a Controller and not a CrudController, how would I utilise the `use \Backpack\CRUD\app\Http\Controllers\Operations\FetchOperation;` operation...if it's even possible?

(FWIW, I tried doing it in a modal but the select2 field seems to have a baked-in z-index that puts it behind the modal instead of in front!! But the relationship field shows up there, or seems to)

Cheers for any help or enlightenment!

**Comment by karandatwani92** [comment]
Hey @fronbow 

What I understand from your text is you are:
-  trying to use Backpack field outside `CrudController`? -  **Backpack fields are not usable outside `CrudController`**
- trying to use Ajax API created by `FetchOperation`? -  **Send a POST request to the AJAX route**.
Note: Only Backpack authenticated users can access this API. That means your custom page should fall under the admin panel.

**Reply by fronbow** [reply]
Sort of, I've created a custom page within the admin side, with this in the blade file
```
    $crud = app()->make('crud');
    $crud->setModel(\App\Models\Notes::class);
    $crud->addField([
        'name' => 'client_id',
        'type' => 'relationship',
        'label' => 'Client',

        'entity' => 'client',
        'attribute' => 'full_name',
        'model' => "App\Models\AllClients",

        'options' => (function ($query){
            return $query
                ->orderby('lastname', 'ASC')->get();
        }),

        'attributes' => [
            'autofocus' => 'on',
        ],
        //'wrapper' => [
        //    'style' => 'z-index: 9999999',
        //],

        'ajax' => true,
        'data_source' => url('fetch/allclients'),

        'default' => $_GET['client_id'] ?? null,
        'allows_null' => true,
        'multiple' => false,
        'hint' => "If you can't find a client, try searching on their surname",
    ]);
```
and `@include('crud::form_content', [ 'fields' => $crud->fields(), 'action' => 'create' ])` at the place where I want the form snippet to appear.
(It was mentioned in a [StackOverflow post](https://stackoverflow.com/questions/68041779/how-to-access-backpack-fields-on-custom-non-crud-page) by @tabacitu from a while ago.

I'm also trying it a different way by utilising a CRUDController with a custom list view to see if that works better.
I'll try the POST request to see if that helps populate my original thinking

Cheers
